### DISCUSSION TITLE
robots.txt - which generated directories should have a Disallow rule?

### DISCUSSION THREAD

**Discussion by mbifulco** [vercel/next.js#31649] [Help]
Hi there,

I've been doing some inspection of my site ([mikebifulco.com](https://mikebifulco.com)'s SEO using the [ahrefs webmaster tools](https://ahrefs.com/webmaster-tools) to occasionally run audits on my site.  In doing so, I noticed that their crawler is checking the _next directory on my site:

![image](https://user-images.githubusercontent.com/1844496/142734344-0a2284d9-4e05-4d9e-bd61-b296d0823ddd.png)

My understanding is that I _probably_ don't want `/_next` and `/api` to be indexed for search, but I haven't found any real guidelines in the docs for next on this. (this is all I found: [static file serving](https://nextjs.org/docs/basic-features/static-file-serving) and ["what is a robots.txt file?"](https://nextjs.org/learn/seo/crawling-and-indexing/robots-txt)

Also, I checked robots.txt for some sites that I thought might show recommended practices... and was a bit surprised.

## vercel.com
[link](https://vercel.com/robots.txt)

```
User-Agent: *
Disallow:
Disallow: /api
Disallow: /oauth
Disallow: /confirm
Disallow: /notifications
Disallow: /old-browser.html
```

## nextjs.org
[link](https://nextjs.org/robots.txt)

```
(404 page)
```

## mikebifulco.com (for comparison)
[link](https://vercel.com/robots.txt)

```
# *
User-agent: *
Allow: /

# Host
Host: https://mikebifulco.com

# Sitemaps
Sitemap: https://mikebifulco.com/sitemap.xml
```

_should I_ add a disallow rule for crawlers on `_next`? What about `/api`, presuming I don't want any of my APIs to show up in search results? Nothing special there, but there's no reason anyone would ever want to browse directly to one of those routes.

**Comment by vpaflah** [comment]
Did you find anything useful?

**Comment by HamedMP** [comment]
Same question here, Claude 3.5 recommends having `_vercel` but not sure if that will impact showing the images on the search results

**Comment by olaj** [comment]
I'm also curious, the AI's (Chat GPT, Gemeni) seems super confidence in that you should block the entire _next folder. But since it also blocks CSS file it seems it can't render the pages (i.e if you do a live test thing in GSC). Like they talk about here.

https://webmasters.stackexchange.com/questions/137447/should-i-block-the-next-directory-in-a-site-using-next-js-with-robots-txt

Maybe it's ok to block all but the .css files or something?

**Comment by fawredd** [comment]
Same question.
